src/test/java/core/basesyntax/service/impl/ParseDataServiceImplTest.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.exception.FruitShopException;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ParseDataServiceImplTest {
+    private static ParseDataServiceImpl parseDataService;
+    private static List<String[]> expectedList;
+
+    @BeforeClass
+    public static void beforeClass() {
+        parseDataService = new ParseDataServiceImpl();
+        expectedList = new ArrayList<>();
+        expectedList.add(new String[]{"d", "e", "f"});
+        expectedList.add(new String[]{"x", "y", "z"});
+    }
+
+    @Test
+    public void parseData_addValidString_Ok() {
+        String input = "a,b,c" + System.lineSeparator() + "d,e,f"
+                + System.lineSeparator() + "x,y,z";
+        List<String[]> actualList = parseDataService.parseData(input);
+        boolean isEqual = true;
+        for (int i = 0; i < actualList.size(); i++) {
+            for (int k = 0; k < actualList.get(i).length; k++) {
+                if (!actualList.get(i)[k].equals(expectedList.get(i)[k])) {
+                    isEqual = false;
+                }
+            }
+        }
+        assertTrue(isEqual);
+    }
+
+    @Test(expected = FruitShopException.class)
+    public void parseData_addInvalidString_notOk() {
+        parseDataService.parseData(null);
+    }
+
+    @Test
+    public void parseData_addIncorrectData_Ok() {
+        expectedList.clear();
+        List<String[]> list = parseDataService.parseData("abc");
+        boolean listLength = list.size() == 0;
+        assertTrue(listLength);

```

Comment:
In this test we expect exception doesn`t it? If no it doesn`t need to change

Reply:
i cutted it